Adding error check of unsupported types

zhenlineo opened the pull request:
Fixes https://github.com/neo4j/neo4j/issues/10239

Commit:
Adding error check of unsupported types

lutovich commented on driver/src/main/java/org/neo4j/driver/v1/Values.java:
```diff
@@ -91,6 +94,10 @@ public static Value value( Object value )
     {
         if ( value == null ) { return NullValue.NULL; }
 
+        if ( value instanceof NodeValue || value instanceof RelationshipValue || value instanceof PathValue )
```
I think it will be better to prohibit node, relationship and path in `PackStreamMessageFormatV1`. It is clear that struct packing code is the single point that writes structs to the channel. While `Values#value()` is just one of many factory methods.

zhenlineo replied:
The reason to do it outside/earlier is because when we got an error there, we will kill the connection, which seems too aggressive. So if we could found the error in advance, then we will simply pop an error to the surface.

lutovich commented on driver/src/test/java/org/neo4j/driver/internal/value/NodeValueTest.java:
```diff
@@ -70,12 +70,12 @@ public void shouldTypeAsNode()
     }
 
-    private NodeValue emptyNodeValue()
+    public static NodeValue emptyNodeValue()
```
Please extract `NodeValueTest#emptyNodeValue()`, `PathValueTest#pathValue()` and `RelationshipValueTest#emptyRelationshipValue()` into a test utility class. Alternatively `ValuesTest` could just create it's own node, relationship and path values.

lutovich reviewed (changes requested):
@zhenlineo review completed. Made two comments. It would also be nice to add ITs which use nodes, relationships and paths as query parameters.

Commit:
Adding more checks around the allowed input parameter types

The input parameters could be of type:
* Map<String, Object>
* Record
* MapValue generated via Values#parameters
* MapValue

If the input parameters are provided by a Map<String, Object> or Record, or Values#parameters, then any unsupported types will be captured via Values#value.

If the input parameters are provided by a MapValue directly, then any unsupported types will be captured at PackStream#pack.

Commit:
Added the server side packing logic for node, rel, and path back in test util
Some tests needs this server side packer to serialize such types.
